### DISCUSSION TITLE
Where can I find the "jars" directory for plugins?

### DISCUSSION THREAD

**Discussion by martina-parkanova** [linkedpipes/etl#989] [Q&A] [answered]
Could you please help me where can I find the "jars" directory for the plugins (components)? I tried to slightly change one of the components and now I don't know where to put its jar file so that it takes effect in my executor. In the documentation, you write about some "jar" directory, which I can't find.. Thanks!

**Comment by skodapetr** [comment, marked as answer]
The directory is set in the property configuration file using [storage.jars.directory property](https://github.com/linkedpipes/etl/blob/develop/deploy/configuration.properties.sample#L118). By default it is in `./deploy/jars` directory. 

The exact location depends on the way you deploy the ETL, especially whether you utilize Docker or not.

**Comment by martina-parkanova** [comment]
Thank you. When we use the existing images, we found the jar files in /opt/lp-etl/components/core (just for the record). But you are right that in the deploy/configuration.properties file, there is "storage.jars.directory = ./jars", so the default directory for our own image probably would be as you said. Unfortunately, I am unable to build my own image, mvn install runs OK, but then docker-compose up crashes on:
![image](https://github.com/linkedpipes/etl/assets/152961774/c4f184d2-d7cf-4960-8d24-c4cad8deeafe)

the directory is there, but probably the script tries to access it frome somewhere else:
![image](https://github.com/linkedpipes/etl/assets/152961774/c6f946c9-dc92-499a-97b6-929e2f905d8a)

Note: I changed the docker-compose file to build the image and not to use it:
  executor:
    # image: ghcr.io/linkedpipes/etl-executor:${LP_VERSION-main}
    build: ./executor

I am building image on windows, but my colleague tried on linux as well, with the same problem.

**Comment by skodapetr** [comment]
Most of the components (executor, frontend, etc. ) need access to whole directory not just the directory of the component. As a result you need to not only set the build directory, but also the Docker file in a slightly different way.

Personally I'm using following on local machine:
```
  storage:
    image: ghcr.io/linkedpipes/etl-storage:${LP_VERSION-main}
    build:
      context: .
      dockerfile: storage/Dockerfile
```
and similar for all components. Could you give it a try?
